csharp/extractor/Semmle.Util/CanonicalPathCache.cs

```diff
@@ -127,8 +127,8 @@ public override string GetCanonicalPath(string path, IPathCache cache)
 
             if (parent != null)
             {
-                string name = Path.GetFileName(path);
-                string parentPath = cache.GetCanonicalPath(parent.FullName);
+                string? name = Path.GetFileName(path);
+                string? parentPath = cache.GetCanonicalPath(parent.FullName);
```

Comment:
Why `?`? `GetCanonicalPath` returns a `string`, not a `string?`.

Reply:
No good reason - fixed.